### ISSUE TITLE
Add semantic headings to all pages

### ISSUE THREAD

**Issue by nwalters512** [PrairieLearn/PrairieLearn#10427]
The lack of semantic headings (`<h1>`, etc.) on some pages has been called out as a potential issue by accessibility reviewers. We should add such headings to ensure that PrairieLearn is meeting accessibility requirements.

On many pages, we use cards that already have headings in the header, they just don't necessarily use `<h1>`/etc. tags. It should be relatively easy to change those all to semantic headings; we just might want some custom CSS for that.

On pages that might not use cards but still need a title for screen readers, we can potentially use screenreader-only titles via Bootstrap's `sr-only` class. This will ensure that folks using screen readers can easily navigate the page, without cluttering it up with unnecessary titles for others.
